server/util/priority_queue/priority_queue.go

```diff
@@ -7,83 +7,118 @@ import (
 	"time"
 )
 
-// A pqItem is the element managed by a priority queue.
-type pqItem[V any] struct {
+// Item is an element managed by a priority queue.
+type Item[V any] struct {
 	value      V
-	priority   int
+	index      int // The index of the item in the heap
+	priority   float64
 	insertTime time.Time
 }
 
-// innerPQ implements heap.Interface and holds pqItems.
-type innerPQ[V any] []*pqItem[V]
+func (i *Item[V]) Value() V {
+	return i.value
+}
+func NewItem[V any](v V, priority float64) *Item[V] {
+	return &Item[V]{
+		value:      v,
+		insertTime: time.Now(),
+		priority:   priority,
+	}
+}
+
+// PriorityQueue implements heap.Interface and holds items.
+type PriorityQueue[V any] []*Item[V]
```

Comment:
It might make sense to rename this to something like `Heap` to make it more clear that this is the "raw" heap, and that `container/heap` operations have to be used on it. The current implementation might be a bit confusing, because `PriorityQueue.Push` appends to the end of the list, while `ThreadSafePriorityQueue.Push` inserts the item at the correct priority position. (Similarly, `PriorityQueue.Pop` removes and returns the last element, while `ThreadSafePriorityQueue.Pop` removes and returns the first element, then restores the heap invariants).

(It might be even clearer to make this private like we had before and have `PriorityQueue` instead just be a non-thread-safe version of `ThreadSafePriorityQueue`, but that seems like a bunch of work and we could maybe do it later)

Reply:
I see your point. The object we create here really just implements the container/heap interface. It's actually not even a heap in the sense that it satisfies the heap property without using heap.Push() or heap.Pop(). 

The go heap documentation has an example where the object was named priorityQueue: https://pkg.go.dev/container/heap#example-package-PriorityQueue. So I will leave it as it is for now.